src/main/java/com/rackspace/ceres/app/services/QueryService.java

```diff
@@ -29,60 +29,80 @@
 import java.time.Duration;
 import java.time.Instant;
 import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.cassandra.core.cql.ReactiveCqlTemplate;
+import org.springframework.data.redis.core.ReactiveRedisTemplate;
+import org.springframework.data.redis.core.ReactiveSetOperations;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 @Service
+@Slf4j
 public class QueryService {
 
   private final ReactiveCqlTemplate cqlTemplate;
   private final MetadataService metadataService;
   private final DataTablesStatements dataTablesStatements;
   private final TimeSlotPartitioner timeSlotPartitioner;
   private final AppProperties appProperties;
+  private ReactiveRedisTemplate<String, List<String>> reactiveRedisTemplate;
 
   @Autowired
   public QueryService(ReactiveCqlTemplate cqlTemplate,
-                      MetadataService metadataService,
-                      DataTablesStatements dataTablesStatements,
-                      TimeSlotPartitioner timeSlotPartitioner,
-                      AppProperties appProperties) {
+      MetadataService metadataService,
+      DataTablesStatements dataTablesStatements,
+      TimeSlotPartitioner timeSlotPartitioner,
+      AppProperties appProperties,
+      ReactiveRedisTemplate reactiveRedisTemplate) {
     this.cqlTemplate = cqlTemplate;
     this.metadataService = metadataService;
     this.dataTablesStatements = dataTablesStatements;
     this.timeSlotPartitioner = timeSlotPartitioner;
     this.appProperties = appProperties;
+    this.reactiveRedisTemplate = reactiveRedisTemplate;
   }
 
-  public Flux<QueryResult> queryRaw(String tenant, String metricName,
+  public Flux<QueryResult> queryRaw(String tenant, String metricKey,
       Map<String, String> queryTags,
       Instant start, Instant end) {
-    return metadataService.locateSeriesSetHashes(tenant, metricName, queryTags)
+    return getMetricGroupFlux(metricKey)
+        .flatMap(
+            metricName -> {
+              String metricNameWithGroup = metricKey+"_"+metricName;
```

Comment:
Hmm.. shouldn't a group be translated into queries with metricNames, I'm not sure I like this underscore thing...

Reply:
taken care.